core/builder/src/main/java/io/quarkus/builder/BuildContext.java

```diff
@@ -240,17 +224,7 @@ private void doProduce(ItemId id, BuildItem value) {
             throw Messages.msg.undeclaredItem(id);
         }
         if (id.isMulti()) {
-            final List<BuildItem> list = execution.getMultis().computeIfAbsent(id, x -> new ArrayList<>());
-            synchronized (list) {
-                if (Comparable.class.isAssignableFrom(id.getType())) {
-                    int pos = Collections.binarySearch((List) list, value);
-                    if (pos < 0)
-                        pos = -(pos + 1);
-                    list.add(pos, value);
-                } else {
-                    list.add(value);
-                }
-            }
+            execution.getMultis().put(stepInfo.getOrdinal(), id, (MultiBuildItem) value);
```

Comment:
This entire sorting behavior was probably a mistake and I'd be glad to see it go away TBH.

Reply:
I found that only `ConfigDescriptionBuildItem` implements `Comparable` in the Quarkus source tree and no Comparable BuildItem in Camel Quarkus. I can have a look if we can eliminate it. WDYT?

Reply:
I think that's a good idea.  It was initially just a random idea but it's not useful in practice.

Reply:
Done in 4810080eb79800718dcd1a42e8c2fba4f5620db1